File: ChatApp.Server/Services/ViewModels/Messages/AdminMessageViewModel.cs
namespace ChatApp.Server.Services.ViewModels.Messages
{
    public class AdminMessageViewModel
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public string SenderUsername { get; set; }
        public int RecipientId { get; set; }
        public string Content { get; set; }
        public bool HasMedia { get; set; }
        public bool IsSeen { get; set; }
        public bool IsDeleted { get; set; }
        public int? ParentMessageId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime ModifiedAt { get; set; }
    }
}

File: chatapp.client/src/app/models/admin-message-view-model.ts
export class AdminMessageViewModel {
  id: number = 0;
  senderId: number = 0;
  senderUsername: string = '';
  recipientId: number = 0;
  content: string = '';
  hasMedia: boolean = false;
  isSeen: boolean = false;
  isDeleted: boolean = false;
  parentMessageId: number | null = null;
  createdAt: string = '';
  modifiedAt: string = '';
}
